perf_test/sparse/KokkosSparse_spgemm.cpp

```diff
@@ -293,11 +291,13 @@ int main (int argc, char ** argv){
     std::cout << "B is not provided. Multiplying AxA." << std::endl;
   }
 
-#if defined( KOKKOS_HAVE_OPENMP )
 
-  if (params.use_openmp) {
-    Kokkos::OpenMP::initialize( params.use_openmp );
+  const int num_threads = params.use_openmp; // Assumption is that use_openmp variable is provided as number of threads
+  const int device_id = params.use_cuda - 1;
+  Kokkos::initialize( Kokkos::InitArguments( num_threads, -1, device_id ) );
 
+#if defined( KOKKOS_HAVE_OPENMP )
+  if (params.use_openmp) {
```

Comment:
again no need for backend specific calls here.

Reply:
Removed the backend-specific print-configuration; left the backend if-guards so appropriate exec space and/or memory space can be set. Would like to move these calls to static functions in structs templated on exec space and mem space, but that will require changing whatever scripts are used for testing this and don't want to break them for now.